pkg/encoding/fix_test.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2017 The Kedge Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package encoding
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/kedgeproject/kedge/pkg/spec"
+
+	api_v1 "k8s.io/client-go/pkg/api/v1"
+)
+
+func TestFixServices(t *testing.T) {
+	failingTest := []spec.ServiceSpecMod{
```

Comment:
@surajssd okay, suppose I want to test the behavior of fixing names, then I'd logically want to put the failing and passing tests together instead of splitting in different functions.
Next whenever I want to add tests for this behavior, I know where to put it because all the tests are together next to each other. If I split the tests, I have to go through both the passing and failing structs to find the relevant tests for the behavior.
It does not make sense to me :(